### ISSUE TITLE
bdist_nuitka produces wrong output for a nested namespace package

### ISSUE THREAD

**Issue by ra-kete** [Nuitka/Nuitka#612]
* Nuitka version, full Python version and Platform (Windows, OSX, Linux ...)

```
$ python3 -m nuitka --version
0.6.7
Python: 3.6.8 (default, Aug 20 2019, 17:12:48)
Executable: /usr/bin/python3
OS: Linux
Arch: x86_64
```

* How did you install Nuitka and Python (pip, anaconda, deb, rpm, from source,
  what is a virtualenv ...), this is very important usually.

```
$ python3 -m pip install nuitka
```

# Issue

I am trying to compile a Python package of mine to an extension module and package it into a wheel using Nuitka. The special thing about my case seems to be that this package resides inside an implicit namespace package nested two levels deep. The compile runs fine, the `.so` file gets created too, but when I install that wheel, my package cannot be imported.

## Minimal example

Consider the following package layout:

```
|- a/b/pkg/__init__.py
|- setup.py
```

`a.b.pkg` is the package I want to compile to an extension module. `a.b` is the implicit namespace package (notice the lack of `__init__.py`s).

These are the contents of `setup.py`:

```python
from setuptools import setup

setup(
    name="pkg",
    packages=["a.b.pkg"],
)
```

I now invoke `bdist_nuitka`:

```
$ python3 setup.py bdist_nuitka
running bdist_nuitka
running build
running build_py
creating build
creating build/lib
creating build/lib/a
creating build/lib/a/b
creating build/lib/a/b/pkg
copying a/b/pkg/__init__.py -> build/lib/a/b/pkg
Building: PyPackage(module_name=a.b.pkg, related_packages=['a.b.pkg'])
Nuitka:INFO:Recurse to import 'a.b.pkg' from 'a/b/pkg'. (Module is exact match of 'a.b.pkg' instructed by user to recurse to.)
Nuitka:INFO:Recurse to import 'a.b' from 'a/b'. (Containing package of recursed module 'a.b.pkg'.)
Nuitka:INFO:Recurse to import 'a' from 'a'. (Containing package of recursed module 'a.b'.)
Nuitka:INFO:Included compiled module 'a'.
Nuitka:INFO:Included compiled module 'a.b'.
Nuitka:INFO:Included compiled module 'a.b.pkg'.
Nuitka:INFO:Included compiled module 'pkg'.
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/a
creating build/bdist.linux-x86_64/wheel/a/b
creating build/bdist.linux-x86_64/wheel/a/b/pkg
creating build/bdist.linux-x86_64/wheel/a.b
copying /tmp/nuitka-test/build/lib/a.b/pkg.pyi -> build/bdist.linux-x86_64/wheel/a.b
copying /tmp/nuitka-test/build/lib/a.b/pkg.so -> build/bdist.linux-x86_64/wheel/a.b
running install_egg_info
running egg_info
creating pkg.egg-info
writing pkg.egg-info/PKG-INFO
writing dependency_links to pkg.egg-info/dependency_links.txt
writing top-level names to pkg.egg-info/top_level.txt
writing manifest file 'pkg.egg-info/SOURCES.txt'
reading manifest file 'pkg.egg-info/SOURCES.txt'
writing manifest file 'pkg.egg-info/SOURCES.txt'
Copying pkg.egg-info to build/bdist.linux-x86_64/wheel/pkg-0.0.0-py3.6.egg-info
creating build/bdist.linux-x86_64/wheel/pkg-0.0.0.dist-info/WHEEL
creating 'dist/pkg-0.0.0-cp36-cp36m-linux_x86_64.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'a.b/pkg.pyi'
adding 'a.b/pkg.so'
adding 'pkg-0.0.0.dist-info/METADATA'
adding 'pkg-0.0.0.dist-info/WHEEL'
adding 'pkg-0.0.0.dist-info/top_level.txt'
adding 'pkg-0.0.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
```

Now, when I install that wheel and try to import my package:

```
$ cd dist
$ pip install pkg-0.0.0-cp36-cp36m-linux_x86_64.whl
Processing ./pkg-0.0.0-cp36-cp36m-linux_x86_64.whl
Installing collected packages: pkg
Successfully installed pkg-0.0.0
$ python3 -c "import a.b.pkg"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'a'
```

My site-packages directory contains the compiled extension module at `a.b/pkg.so`. I think this is the reason why the import doesn't work. Shouldn't it be at `a/b/pkg.so` instead?

**Comment by kayhayen** [comment]
@tommyli3318 Can you have a look at this one, seems easy and where you made changes previously. The `a.b` is wrong, and we probably just need to handle multiple levels. Namespace packages for distutils are supported, but maybe only one level deep?

**Comment by ra-kete** [comment]
Fixed in #646.

**Comment by kayhayen** [comment]
The release 0.6.8 was just made and contains the correction.
